src/components/lib/posix/posix.c

```diff
@@ -154,7 +148,7 @@ void *
 cos_mremap(void *old_address, size_t old_size, size_t new_size, int flags)
 {
 	printc("mremap not implemented!\n");
-	return -ENOTSUP;
+	return MAP_FAILED;
```

Comment:
Because of the function return type. `ENOTSUP` is `int` and ret type is `void*`.